plugins/woocommerce/includes/bis/class-wc-bis-templates.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * WC_BIS_Templates class
+ *
+ * @package  WooCommerce Back In Stock Notifications
+ * @since    1.0.0
+ */
+
+// Exit if accessed directly.
+use Automattic\Jetpack\Constants;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Display functions and filters.
+ *
+ * @class    WC_BIS_Templates
+ * @version  1.6.7
+ */
+class WC_BIS_Templates {
+
+	/**
+	 * Should dequeue scripts status.
+	 *
+	 * @var array
+	 */
+	private $should_dequeue_scripts;
+
+	/**
+	 * Setup hooks and functions.
+	 */
+	public function __construct() {
+
+		// Template functions and hooks.
+		require_once WC_ABSPATH . 'includes/bis/wc-bis-template-functions.php';
+		require_once WC_ABSPATH . 'includes/bis/wc-bis-template-hooks.php';
+
+		// Front end scripts and JS templates.
+		add_action( 'wp_enqueue_scripts', array( $this, 'frontend_scripts' ) );
+		add_action( 'wp_print_footer_scripts', array( $this, 'dequeue_product_scripts' ), 9 );
+
+		// Defaults.
+		$this->should_dequeue_scripts = true;
+	}
+
+	/*
+	---------------------------------------------------*/
+	/*
+		Setters.                                         */
+	/*---------------------------------------------------*/
+
+	/**
+	 * Force enqueuing scripts or not.
+	 *
+	 * @return void
+	 */
+	public function enqueue_scripts() {
+		$this->should_dequeue_scripts = false;
+	}
+
+	/*
+	---------------------------------------------------*/
+	/*
+		Callbacks.                                       */
+	/*---------------------------------------------------*/
+
+	/**
+	 * Front-end styles and scripts.
+	 *
+	 * @return void
+	 */
+	public function frontend_scripts() {
+
+		$suffix  = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
+		$version = Constants::get_constant( 'WC_VERSION' );
+
+		// Styles
+		wp_register_style( 'wc-bis-css', WC()->plugin_url() . '/assets/css/frontend/bis.css', false, $version, 'all' );
+		wp_style_add_data( 'wc-bis-css', 'rtl', 'replace' );
+		wp_enqueue_style( 'wc-bis-css' );
```

Comment:
Something else to consider is that we load a single CSS for each frontend view. If a block theme is active, then we also load another one. Honestly, regarding this merge, I think it would be better to: 
- Repurpose the `dequeue_product_scripts` function to also handle styles, 
- Merge the block theme with the main CSS (possibly by prefixing rules with the `woocommerce-uses-block-theme` class).

WDYT? @peterfabian

Reply:
Sounds like a good suggestion. Something like this perhaps? [458101e](https://github.com/woocommerce/woocommerce/pull/55787/commits/458101e0f5baa15d681bbbe15390e010d0629db6)

Reply:
Looking good! Left a comment here https://github.com/woocommerce/woocommerce/pull/55787/commits/458101e0f5baa15d681bbbe15390e010d0629db6#r1995311931